File: second_year/projects/G-OOP-400-COT-4-1-bsraytracer-13/src/rectangle3D.cpp
/*
** EPITECH PROJECT, 2026
** snd_year_projects
** File description:
** rectangle3D
*/

#include "../include/rectangle3D.hpp"
using namespace Math;
using namespace RayTracer;

Rectangle3D::Rectangle3D(Point3D givenOrigin, Vector3D givenBotton, Vector3D givenLeft)
    : origin(givenOrigin), bottom_side(givenBotton), left_side(givenLeft)
{}

Point3D Rectangle3D::pointAt(double u, double v)
{
    if (u < 0 || u > 1 || v < 0 || v > 1 )
        return Point3D(-1.0, -1.0, -1.0);
    return origin + (bottom_side * u) + (left_side * v);
}

File: second_year/projects/G-OOP-400-COT-4-1-bsraytracer-13/rust/src/rectangle3d.rs
//
// EPITECH PROJECT, 2026
// snd_year_projects
// File description:
// rectangle3D
//

use crate::Vector3D;
use crate::Point3D;

pub struct Rectangle3D {
    pub origin: Point3D,
    pub left_side: Vector3D,
    pub bottom_side: Vector3D,
}

impl Rectangle3D {
    pub fn pointAt(&self, u: f32, v: f32) -> Point3D {
        return self.origin + (self.bottom_side * u) + (self.left_side * v);
    }
}
